code/modules/mob/living/simple_animal/hostile/megafauna/chaos_marine.dm

```diff
@@ -0,0 +1,510 @@
+	name = "Infernal Summon"
+	icon_icon = 'icons/mob/mob.dmi'
+	button_icon_state = "imp"
+	chosen_message = "<span class='colossus'>You will now summon demons to assist you.</span>"
+	chosen_attack_num = 4
+
+/datum/action/innate/megafauna_attack/teleport_b
+	name = "Teleport"
+	icon_icon = 'icons/obj/projectiles.dmi'
+	button_icon_state = "bluespace"
+	chosen_message = "<span class='colossus'>You will now teleport next to your target.</span>"
+	chosen_attack_num = 5
+
+/datum/action/innate/megafauna_attack/rapid_fire
+	name = "Rapid Fire"
+	icon_icon = 'icons/obj/guns/projectile.dmi'
+	button_icon_state = "arg"
+	chosen_message = "<span class='colossus'>You are now rapidly shooting at your enemy with explosive bullets (5).</span>"
+	chosen_attack_num = 6
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/AttackingTarget()
+	if(charging)
+		return
+	face_atom(target)
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			visible_message("<span class='danger'>[src] butchers [L]!</span>",
+			"<span class='userdanger'>You butcher [L], restoring your health!</span>")
+			if(!is_station_level(z) || client) //NPC monsters won't heal while on station
+				adjustHealth(-(L.maxHealth * 0.5))
+			L.gib()
+			if(prob(50))
+				telegraph()
+				say(pick("Blood for the blood god!", "Skulls for the skull throne!", "Blood! Blood!!", "Die! Die! Die!", "You will be a fine offering!", "Chaos!"))
+			return TRUE
+	weapon.melee_attack_chain(src, target)
+	return TRUE
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/Move()
+	if(charging)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/proc/adjustCMspeed()
+	if(health <= maxHealth*0.25)
+		speed = 4
+		dash_mod = 0.4
+		dash_num = 5
+		rapid_melee = 3.5
+	else if(health <= maxHealth*0.5)
+		speed = 4.8
+		dash_mod = 0.6
+		dash_num = 4
+		rapid_melee = 2.5
+	else if(health <= maxHealth*0.75)
+		speed = 5.4
+		dash_mod = 0.8
+		dash_num = initial(dash_num)
+		rapid_melee = 1.5
+	else if(health > maxHealth*0.75)
+		speed = initial(speed)
+		dash_mod = initial(dash_mod)
+		dash_num = initial(dash_num)
+		rapid_melee = initial(rapid_melee)
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/do_attack_animation(atom/A, visual_effect_icon, obj/item/used_item, no_effect)
+	if(!used_item && !isturf(A))
+		used_item = weapon
+	..()
+
+/mob/living/simple_animal/hostile/megafauna/chaos_marine/OpenFire()
+	adjustCMspeed()
+
+	if(client)
+		switch(chosen_attack)
+			if(1)
+				blood_dash(target)
+			if(2)
+				runic_blast()
+			if(3)
+				blast()
+			if(4)
+				infernal_summon()
+			if(5)
+				teleport_b(target)
+			if(6)
+				rapid_fire()
+		return
+
+	Goto(target, move_to_delay, minimum_distance)
+	if(get_dist(src, target) >= 1 && dash_cooldown <= world.time && !charging)
+		blood_dash(target)
+	if(get_dist(src, target) > round(5*dash_mod) && teleport_cooldown <= world.time && !charging)
```

Comment:
should these be else ifs, or is it intended that a chaos space marine will start combats by using like 3 moves at once?

Reply:
It is supposed to be full random, given that you are in range and not currently using another attack.